### PR TITLE
Add contrastive loss 

### PR DESCRIPTION
**Pull request by vishal-14069**
## Proposed changes

This pull request adds support for computing contrastive loss between input vector pairs in the loss modules.

Test ground truth values are verified by TensorFlow and a PyTorch function.

TensorFlow Implementation: https://www.tensorflow.org/addons/api_docs/python/tfa/losses/ContrastiveLoss

## Checklist
Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Comment by vishal-14069** [comment]
Hello @awni can you please review this PR, I accidentally forgot to add reviewers.

**Review by Stealeristaken — APPROVED** [review]

**Comment by awni** [comment]
Hi @vishal-14069 thanks for the contribution. We already decided in #324 not to add a contrastive loss as:

1. It's not that hard to implement
2. It's likely people want to specialize it to different use cases

FYI for the future if you are unsure whether to make a contribution, feel free to check the issues and/or start an issue / discussion first.

**Comment by vishal-14069** [comment]
Thanks for the review @Stealeristaken. Yes @awni makes sense, I will do that from now on. Thank you!
